### PR TITLE
Csfun tree exercises

### PR DESCRIPTION
**Pull request by CheezItMan**
## Description

Some in-class questions.

## Feedback Requested
Please utilize the built-in GitHub Reviewers functionality to assign necessary reviewers. Use this section to further specify portions of the PR that you would like additional or focused feedback on.

### CODE REVIEW

File: 04-cs-fundamentals/internship/exercises/binary-tree.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# Binary Tree Exercises
+
+## Instructions
+
+In pairs answer the following questions as if they are whiteboarding exercises.  Write down your answers for each step.
+
+For Each Question:
+
+- **Step 1:**  Write down 2 - 3 clarifying questions you would ask an interviewer.
```
Comment (kaidamasaki) [inline 04-cs-fundamentals/internship/exercises/binary-tree.md]: Should these line up with PSEs?

File: 04-cs-fundamentals/internship/exercises/binary-tree.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+class Student
+  attr_reader :id, :name
+
+  def initialize(id, name)
+    @id = id
+    @name = name
+  end
+end
+
+class TreeNode
```
Comment (kaidamasaki) [inline 04-cs-fundamentals/internship/exercises/binary-tree.md]: I'd probably name the class `Tree` here since it's a complete data structure on its own.

File: 04-cs-fundamentals/internship/exercises/binary-tree.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  attr_reader :key, :value
+  attr_accessor :left, :right
+
+  def initialize(k, v, l = nil, r = nil)
```
Comment (kaidamasaki) [inline 04-cs-fundamentals/internship/exercises/binary-tree.md]: Having full argument names would be nice.

Assuming you want to be nice, that is.

File: 04-cs-fundamentals/internship/exercises/binary-tree.md (hunk 1 above)
Comment (CheezItMan) [reply 04-cs-fundamentals/internship/exercises/binary-tree.md]: Mostly they do, there are fewer clarifying questions because largely C11 is pretty good at that, and coming up with edge cases to test their code against, is something they struggle more with.

File: 04-cs-fundamentals/internship/exercises/binary-tree.md (hunk 2 above)
Comment (CheezItMan) [reply 04-cs-fundamentals/internship/exercises/binary-tree.md]: It parallels the Tree exercises we've used at Ada and are in the Cracking the Coding Interview book.  So sticking with what's familiar.
